diff --git a/node_modules/react-native-unistyles/src/web/shadowRegistry.ts b/node_modules/react-native-unistyles/src/web/shadowRegistry.ts
--- a/node_modules/react-native-unistyles/src/web/shadowRegistry.ts
+++ b/node_modules/react-native-unistyles/src/web/shadowRegistry.ts
@@ -75,2 +75,7 @@ export class UnistylesShadowRegistry {
-            return deepMergeObjects(...allStyles)
+            const dependencies = Array.from(new Set(allStyles.flatMap(extractUnistyleDependencies)))
+
+            return {
+                ...deepMergeObjects(...allStyles),
+                ...(dependencies.length > 0 ? { uni__dependencies: dependencies } : {}),
+            }
         }
diff --git a/node_modules/react-native-unistyles/lib/module/web/shadowRegistry.js b/node_modules/react-native-unistyles/lib/module/web/shadowRegistry.js
--- a/node_modules/react-native-unistyles/lib/module/web/shadowRegistry.js
+++ b/node_modules/react-native-unistyles/lib/module/web/shadowRegistry.js
@@ -68,2 +68,8 @@ export class UnistylesShadowRegistry {
-      return deepMergeObjects(...allStyles);
+      const dependencies = Array.from(new Set(allStyles.flatMap(extractUnistyleDependencies)));
+      return {
+        ...deepMergeObjects(...allStyles),
+        ...(dependencies.length > 0 ? {
+          uni__dependencies: dependencies
+        } : {})
+      };
     };
diff --git a/node_modules/react-native-unistyles/lib/commonjs/web/shadowRegistry.js b/node_modules/react-native-unistyles/lib/commonjs/web/shadowRegistry.js
--- a/node_modules/react-native-unistyles/lib/commonjs/web/shadowRegistry.js
+++ b/node_modules/react-native-unistyles/lib/commonjs/web/shadowRegistry.js
@@ -72,2 +72,8 @@ class UnistylesShadowRegistry {
-      return (0, _utils.deepMergeObjects)(...allStyles);
+      const dependencies = Array.from(new Set(allStyles.flatMap(_utils2.extractUnistyleDependencies)));
+      return {
+        ...(0, _utils.deepMergeObjects)(...allStyles),
+        ...(dependencies.length > 0 ? {
+          uni__dependencies: dependencies
+        } : {})
+      };
     };
